src/main/java/com/mathworks/ci/utilities/MatlabCommandRunner.java

```diff
@@ -0,0 +1,215 @@
+
+        // Add custom environment vars
+        EnvVars env = getEnvVars();
+        Utilities.addMatlabToEnvPathFrmAxis(
+                Computer.currentComputer(), 
+                this.params.getTaskListener(), 
+                env);
+
+        ProcStarter proc = this.params.getLauncher().launch()
+            .envs(env)
+            .cmds(args);
+        if (this.stdOut == null) {
+            proc.stdout(this.params.getTaskListener());
+        } else {
+            proc.stdout(this.stdOut);
+        }
+        proc.pwd(this.params.getWorkspace());
+
+        int code = proc.join();
+
+        if (code != 0) {
+            throw new MatlabExecutionException(code);
+        }
+    }
+
+    /**
+     * Redirects stdout.
+     *
+     * @param out the OutputStream to write to
+     */
+    public void redirectStdOut(OutputStream out) {
+        this.stdOut = out;
+    }
+
+    /**
+     * Adds an environment variable.
+     *
+     * @param key the environment variable name
+     * @param value the environment variable value
+     */
+    public void addEnvironmentVariable(String key, String value) {
+        additionalEnvVars.put(key, value);
+    }
+
+    public EnvVars getEnvVars() {
+        EnvVars env = new EnvVars(this.params.getEnvVars());
+        env.putAll(additionalEnvVars); 
+        return env;
+    }
+
+    /**
+     * Copies a resource into the temporary folder.
+     *
+     * @param sourceFile the name of a resource the class loader can find.
+     * @param targetFile the name of the file to create in the temp folder.
+     * @return the FilePath to the new location in the temp folder.
+     */
+    public FilePath copyFileToTempFolder(String sourceFile, String targetFile) throws IOException, InterruptedException {
+        final ClassLoader classLoader = getClass().getClassLoader();
+        FilePath targetFilePath = new FilePath(this.tempFolder, targetFile);
+        InputStream in = classLoader.getResourceAsStream(sourceFile);
+        targetFilePath.copyFrom(in);
+        targetFilePath.chmod(0755);
+
+        return targetFilePath;
+    }
+
+    public FilePath getTempFolder() {
+        return tempFolder;
+    }
+
+    /**
+     * Creates a file with the specified content in the temporary folder.
+     *
+     * Additionally, the file content will be prefixed with a statement returning to the MATLAB starting folder.
+     *
+     * @param content string that represents the content of the file.
+     * @return the FilePath to the script file that is created.
+     */
+    protected FilePath createFileWithContent(String content) throws IOException, InterruptedException {
+        String fileName = "script_" + RandomStringUtils.randomAlphanumeric(8) + ".m";
+        FilePath scriptFile = new FilePath(this.tempFolder, fileName);
+
+        String expandedContent = getEnvVars().expand(content);
+        String finalContent = "cd(getenv('MW_ORIG_WORKING_FOLDER'));\n"
+            + expandedContent;
```

Comment:
I don't understand the expanded content here. Can you help give an example of what it will look like with and without environment variables?

Reply:
Yeah!

In the following example, the user entered the literal string `"${TASKS}"` into the tasks UI textbox, and nothing into the build options textbox.

The user also has set the environment variable `TASKS` to be `clean lint test` on their build agent.

Result:
```
contentBeforeExpansion = "buildtool ${TASKS}"
contentAfterExpansion = "buildtool clean lint test"
```